pkg/test_framework/test.go

```diff
@@ -0,0 +1,153 @@
+package framework
+
+import (
+	"fmt"
+	"log"
+	"os/exec"
+	"sync/atomic"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	harness "github.com/dlespiau/kube-test-harness"
+)
+
+type Test struct {
+	harness.Test
+
+	stopOperator bool
+	harness *Harness
+}
+
+func (t *Test) Setup() *Test {
+	_ = t.Test.Setup()
+	return t
+}
+
+func (t *Test) StartOperator() error {
+	if t.stopOperator == true {
+		return fmt.Errorf("operator already started")
+	}
+	err := startOperator(t.harness.Options, t.harness.Sinks)
+	if err == nil {
+		t.stopOperator = true
+	}
+	return err
+}
+
+func (t *Test) StopOperator() {
+	if t.stopOperator {
+		stopOperator(t.harness.Options, t.harness.Sinks)
+		t.stopOperator = false
+	}
+}
+
+func (t *Test) DeleteDeployment(d *appsv1.Deployment, timeout time.Duration) {
+	t.Test.DeleteDeployment(d)
+	t.Test.WaitForDeploymentDeleted(d, timeout)
```

Comment:
probably useful to return timeout error, if timed out

Reply:
I have to rely on kube-test-harness for such functionality - and it will fail test if timeout occurred.